tests/EnumDynamicReturnTypeExtensionGetValuesTest.php

```diff
@@ -0,0 +1,158 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MabeEnumPHPStanTest;
+
+use MabeEnum\Enum;
+use MabeEnumPHPStan\EnumDynamicReturnTypeExtension;
+
+final class EnumDynamicReturnTypeExtensionGetValuesTest extends ExtensionTestCase
+{
+    /** @var EnumDynamicReturnTypeExtension */
+    private $extension;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        if (!method_exists(Enum::class, 'getValues')) {
+            $this->markTestSkipped('Enum::getValues() not supported in version 1.x');
+        }
+
+        $this->extension = new EnumDynamicReturnTypeExtension();
+    }
+
+    public function testNullType(): void
+    {
+        $code = <<<'CODE'
+<?php
+function f(MabeEnumPHPStanTest\Assets\NullTypeEnum $enum) {
+    die;
+}
+CODE;
+
+        $this->processCode($code, '$enum->getValues()', 'array(null)', $this->extension);
+    }
+
+    public function testBoolType(): void
+    {
+        $code = <<<'CODE'
+<?php
+function f(MabeEnumPHPStanTest\Assets\BoolTypeEnum $enum) {
+    die;
+}
+CODE;
+
+        $this->processCode($code, '$enum->getValues()', 'array(true, false)', $this->extension);
+    }
+
+    public function testStringType(): void
+    {
+        $code = <<<'CODE'
+<?php
+function f(MabeEnumPHPStanTest\Assets\StrTypeEnum $enum) {
+    die;
+}
+CODE;
+
+        $this->processCode($code, '$enum->getValues()', "array('str1', 'str2')", $this->extension);
+    }
+
+    public function testIntType(): void
+    {
+        $code = <<<'CODE'
+<?php
+function f(MabeEnumPHPStanTest\Assets\IntTypeEnum $enum) {
+    die;
+}
+CODE;
+
+        $this->processCode($code, '$enum->getValues()', 'array(0, 1)', $this->extension);
+    }
+
+    public function testFloatType(): void
+    {
+        $code = <<<'CODE'
+<?php
+function f(MabeEnumPHPStanTest\Assets\FloatTypeEnum $enum) {
+    die;
+}
+CODE;
+
+        $this->processCode($code, '$enum->getValues()', 'array(1.1, 1.2)', $this->extension);
+    }
+
+    public function testArrayType(): void
+    {
+        $code = <<<'CODE'
+<?php
+function f(MabeEnumPHPStanTest\Assets\ArrayTypeEnum $enum) {
+    die;
+}
+CODE;
+
+        $this->processCode($code, '$enum->getValues()', 'array(array(array()))', $this->extension);
+    }
+
+    public function testAllTypes(): void
+    {
+        $code = <<<'CODE'
+<?php
+function f(MabeEnumPHPStanTest\Assets\AllTypeEnum $enum) {
+    die;
+}
+CODE;
+
+        $this->processCode(
+            $code,
+            '$enum->getValues()',
+            "array(null, true, 1, 1.1, 'str', array(null, true, 1, 1.1, 'str', array()))",
+            $this->extension
+        );
+    }
+
+    public function testBaseEnum(): void
+    {
+        $code = <<<'CODE'
+<?php
+function f(MabeEnum\Enum $enum) {
+    die;
+}
+CODE;
+
+        $this->processCode($code, '$enum->getValues()', 'array', $this->extension);
```

Comment:
This may be empty array? aka `array()` if i understand difference between array and array() correectly

Reply:
No, this means `$enum` is an enumerator of an unknown enumeration and the known return type of `getValues` is just `array`.
On the other site `array()` means a `ConstantArrayType` of an empty array, in this case you would require an enumerator of an enumeration with no enumerators which is impossible.

Reply:
ok, i over complicated this.
can you pls add empty enum to test cases as well?